src/librustc/ty/layout.rs

```diff
@@ -863,6 +869,11 @@ impl<'tcx> LayoutCx<'tcx, TyCtxt<'tcx>> {
                     return Ok(tcx.intern_layout(st));
                 }
 
+                // At this point, we have handled all unions and
+                // structs. (We have also handled univariant enums
+                // that allow niche-optimization.)
```

Comment:
Not sure why niches are mentioned, it's all univariant enums which can be "optimized" (irrespective of niches), i.e. they don't have `#[repr(C)]` nor `#[repr(iN)]` attributes which mandate a certain tagged `enum` layout.

Reply:
fair enough, I'll `sed -e s/niche-optimization/representation optimization/` here. I just didn't want to accidentally imply that *all* enums are handled in the remaining code.